### DISCUSSION TITLE
How about interception of $query-> get() results ? any hooks/events?

### DISCUSSION THREAD

**Discussion by LINKeRxUA** [yajra/laravel-datatables#2627] [Ideas]
Hello! I can't find any solution to solve some problems. 
I have some method that retrieves from database different permissions.

Using datatable method **addColumn** i'm checking about abilities of execution!
```
->addColumn('batch', function($row){
                return [
                    self::BATCH_DELETE => $this->checkIfCan('cmsDelete', $row),
                ];
            })
```
my method **checkIfCan** is executed for **N** times, and each time it executes "select .... where id = $id" query.  100 times per 100 rows
I want to retrieve all permissions using "select .... where id in (1,2...,N)" before datatable json content is generated and after it executes the query.

Is there any ability to intercept all rows before building json? (like with laravel chunk method, which passes all rows into closure)

```
$datatable
->of(...)
->addColumn()
->....
->onDataRetrieved(function($allRows){ ...my code to get permissions... }) // need something like this
->toJson()

```

**Comment by yajra** [comment]
We don't have any event hooks for this yet afaik. I like this idea of `onDataRetrieved`, would you be able to submit a PR for this? Will try to implement when I got the chance. Thanks!

**Reply by LINKeRxUA** [reply]
Sure! I should finish my project first, and then i'll add callback features to pull request!
